File: _GTW/media/js/GTW/jQ/Image_Selection.js
// Copyright (C) 2012-2013 Martin Glueck All rights reserved
// Langstrasse 4, A--2244 Spannberg, Austria. martin@mangari.org
// #*** <License> ************************************************************#
// This script is licensed under the terms of either the
// MIT License or the GNU Affero General Public License (AGPL) Version 3.
// http://www.c-tanzer.at/license/mit_or_agpl.html
// #*** </License> ***********************************************************#
//
//++
// Name
//    GTW/jQ/Image_Selection
//
// Purpose
//    Special code for the handling of the image url selection
//
// Revision Dates
//    17-Aug-2012 (MG) Creation
//    17-Aug-2012 (MG) Make selection window url configurable
//    24-Jan-2013 (MG) Use elfinder as media selection interface
//    28-Jan-2013 (MG) `_setup_field` now has two parameters
//    ««revision-date»»···
//--

"use strict";

( function ($) {
    var Image_Selection_Field = $GTW.AFS.Elements.Field.extend (
        { _setup_field : function _setup_field (inp$, Form) {
            var selection_url = this.selection_url;
            var url = inp$.val ();
            var img$ = $("<img />").attr ({ src : url, alt : this.label})
                                  .addClass ("afs-media-button");
            inp$.prop   ("type", "hidden")
                .after  (img$);
            var elfinder = $('<div />').dialogelfinder ({
                      url             : this.elfinder
                    , lang            : this.language
                    , width           : 840
                    , destroyOnClose  : false
                    , getFileCallback : function(files, fm) {
                          img$.attr ("src", files);
                          inp$.val  (files).trigger ("change");
                          console.log (files);
                      }
                    , commandsOptions : {
                        getfile : {
                              oncomplete : 'close'
                            , folders    : true
                        }
                      }
                    , autoOpen : false
                });
            img$.click (function (ev) {
                var this$ = $(this);
                window.SetUrl = function (url) {
                };
                elfinder.dialogelfinder ("open");
            });
          }
        }
      , { type_name : "Image_Selection_Field" }
    );

    $GTW.AFS.Elements.Image_Selection_Field = Image_Selection_Field;
  } (jQuery)
);

// __END__ GTW/jQ/Image_Selection.js

File: _GTW/_AFS/_MOM/Image_Selection.py
# -*- coding: utf-8 -*-
# Copyright (C) 2013 Martin Glueck All rights reserved
# Langstrasse 4, A--2244 Spannberg, Austria. martin@mangari.org
# #*** <License> ************************************************************#
# This module is part of the package GTW.AFS.MOM.
# 
# This module is licensed under the terms of the BSD 3-Clause License
# <http://www.c-tanzer.at/license/bsd_3c.html>.
# #*** </License> ***********************************************************#
#
#++
# Name
#    GTW.AFS.MOM.Image_Selection
#
# Purpose
#    Special field for adding support for the Image_Selection WYSIWYG editor.
#
# Revision Dates
#    29-Jan-2013 (MG) Creation
#--

from   __future__ import absolute_import, division, print_function, unicode_literals

from   _GTW                     import GTW
from   _GTW._AFS._MOM           import Element
from   _JNJ                     import JNJ
import _JNJ.Templateer
from   _TFL.I18N                import _T

JNJ.Template \
    ( "afs_img_selection"
    , "html/AFS/img_selection.jnj"
    , parent_name = "afs_div_seq"
    )


class Image_Selection_Field (Element.Field) :
    """A field which allows the selection of an image"""

    renderer     = "afs_img_selection"

    def __init__ (self, * args, ** kw) :
        kw.setdefault ("elfinder", "/elfinder")
        kw.setdefault ("language", "en")
        self.__super.__init__ (* args,  ** kw)
    # end def __init__

# end class Image_Selection_Field

if __name__ != "__main__" :
    GTW.AFS.MOM._Export ("*")
### __END__ GTW.AFS.MOM.Image_Selection
